databroker/src/grpc/kuksa_val_v2/conversions.rs

```diff
@@ -215,94 +217,170 @@ impl From<&proto::Datapoint> for broker::DataValue {
-                                },
-                            )),
-                        })
-                    }
-                    broker::DataValue::Uint32Array(vec) => {
-                        Some(proto::ValueRestriction {
-                            r#type: Some(proto::value_restriction::Type::Unsigned(
-                                proto::ValueRestrictionUint {
-                                    allowed_values: vec.iter().cloned().map(u64::from).collect(),
-                                    min: None, // TODO: Implement
-                                    max: None, // TODO: Implement
-                                },
-                            )),
-                        })
-                    }
-                    broker::DataValue::Uint64Array(vec) => {
-                        Some(proto::ValueRestriction {
-                            r#type: Some(proto::value_restriction::Type::Unsigned(
-                                proto::ValueRestrictionUint {
-                                    allowed_values: vec.clone(),
-                                    min: None, // TODO: Implement
-                                    max: None, // TODO: Implement
-                                },
-                            )),
-                        })
-                    }
-                    broker::DataValue::FloatArray(vec) => {
-                        Some(proto::ValueRestriction {
-                            r#type: Some(proto::value_restriction::Type::FloatingPoint(
-                                proto::ValueRestrictionFloat {
-                                    allowed_values: vec.iter().cloned().map(f64::from).collect(),
-                                    min: None, // TODO: Implement
-                                    max: None, // TODO: Implement
-                                },
-                            )),
-                        })
-                    }
-                    broker::DataValue::DoubleArray(vec) => {
-                        Some(proto::ValueRestriction {
-                            r#type: Some(proto::value_restriction::Type::FloatingPoint(
-                                proto::ValueRestrictionFloat {
-                                    allowed_values: vec.clone(),
-                                    min: None, // TODO: Implement
-                                    max: None, // TODO: Implement
-                                },
-                            )),
-                        })
-                    }
+                    });
+                };
+            }
+            DataType::Int8
+            | DataType::Int16
+            | DataType::Int32
+            | DataType::Int64
+            | DataType::Int8Array
+            | DataType::Int16Array
+            | DataType::Int32Array
+            | DataType::Int64Array => {
+                let min_value = match metadata.min {
+                    Some(DataValue::Int32(value)) => Some(i64::from(value)),
```

Comment:
Is it not possible in VSS to have `Int8` data types with `Int8` `min` and `max`?

Reply:
Indeed.

But In Databroker we never use smaller datatypes than 32 bits for integers, see https://github.com/eclipse-kuksa/kuksa-databroker/blob/81a15f1db4d0bdb25d69233bd9bba7e0693c4d51/databroker/src/types.rs#L59

The general Databroker approach (as I understand it) is to use the VSS Type as basis for checking min/max, but then store it internally in i32/i64/u32/u64.

Reply:
> The general Databroker approach (as I understand it) is to use the VSS Type as basis for checking min/max, but then store it internally in i32/i64/u32/u64.

Exactly